tests/integration/tier0/check-internet-connection/test_internet_connection.py

```diff
@@ -0,0 +1,78 @@
+import os
+import shutil
+
+from envparse import env
+
+
+# The original dnsmasq.conf file
+DNSMASQ_CONF_FILE = "/etc/dnsmasq.conf"
+
+# The original resolv.conf file
+RESOLV_CONF_FILE = "/etc/resolv.conf"
+
+# The backup file for dnsmasq, should be /etc/dnsmasq.conf.bk
+DNSMASQ_CONF_BACKUP_FILE = "{0}.bk".format(DNSMASQ_CONF_FILE)
+
+# The backup file for resolv, should be /etc/resolv.conf.bk
+RESOLV_CONF_BACKUP_FILE = "{0}.bk".format(RESOLV_CONF_FILE)
+
+
+def _modify_configuration_files():
+    """Modify the configuration files and backup the original ones."""
+    # Backup the dnsmasq and resolv files
+    shutil.copy(DNSMASQ_CONF_FILE, DNSMASQ_CONF_BACKUP_FILE)
+    shutil.copy(RESOLV_CONF_FILE, RESOLV_CONF_BACKUP_FILE)
+
+    with open("/etc/dnsmasq.conf", "a") as f:
+        # Everything else is resolved to localhost
+        f.write("address=/#/127.0.0.1\n")
+
+    with open("/etc/resolv.conf", "a") as f:
+        f.write("nameserver 127.0.0.1\n")
+
+
+def _restore_configuration_files():
+    """Restore the original files and remove the modified ones."""
+    if os.path.exists(DNSMASQ_CONF_FILE):
+        os.remove(DNSMASQ_CONF_FILE)
+        shutil.move(DNSMASQ_CONF_BACKUP_FILE, DNSMASQ_CONF_FILE)
+
+    if os.path.exists(RESOLV_CONF_FILE):
+        os.remove(RESOLV_CONF_FILE)
+        shutil.copy(RESOLV_CONF_BACKUP_FILE, RESOLV_CONF_FILE)
+
+
+def test_check_if_internet_connection_is_reachable(convert2rhel):
+    """Test if convert2rhel can access the internet."""
+    with convert2rhel(
+        ("-y --no-rpm-va --serverurl {} --username {} --password {} --pool {} --debug").format(
+            env.str("RHSM_SERVER_URL"),
+            env.str("RHSM_USERNAME"),
+            env.str("RHSM_PASSWORD"),
+            env.str("RHSM_POOL"),
+        )
+    ) as c2r:
+        c2r.expect("Checking internet connectivity using address")
+        assert c2r.expect_exact("Internet connection available.") == 0
+        c2r.send(chr(3))
+
+    assert c2r.exitstatus == 1
```

Comment:
The c2r returned exic code 130.
https://privatebin.corp.redhat.com/?65fd22902d1c97a8#BKzohheYwPpGQ8xkMf4G6vwk975CyKstj9Ne5pexeNhF
I'm not sure why, maybe because it's right in the beginning of the conversion.

Maybe the safest option here would be to use:?
```suggestion
    assert c2r.exitstatus != 0
```

Reply:
I'm guessing that's because we are sending SIGTERM right when we are executing the systemctl call to assert that dbus is present. Let me take a new look.

Reply:
@r0x0d Sadly I don't think there is a faster "way". It's caused by some delay during processing the line during the c2r run and reacting to it.

You can try to run the interactive way (without the `-y` option). Since it's so early in the conversion, you can then send the `n` to the first prompt. This seems like a more clean way and we should always get the `1` exit code.

Also you will not need to pass the credentials to the call, simple 
```
with convert2rhel(
        ("--no-rpm-va --debug")
        ...
```
should be enough.

Reply:
Okay, seems a better approach indeed. I will adapt the tests to that. Thanks!

Reply:
Done!